src/models/booking.cr

```diff
@@ -69,6 +119,17 @@ class Booking
     booking_model.approver_email = booking_model.approver_email if booking_model.approver_email_column.defined?
     booking_model.email_digest = booking_model.user_email.digest
     booking_model.booked_by_email_digest = booking_model.booked_by_email.digest
+    booking_model.booked_from = booking_model.utm_source if !booking_model.booked_from_column.defined?
+    booking_model.history = booking_model.current_history
+  end
+
+  def current_history : Array(History)
+    state = current_state
+    history_column.value([] of History).dup.tap do |booking_history|
+      if booking_history.empty? || booking_history.last.state != state
+        booking_history << History.new(state, Time.local.to_unix, @utm_source) unless state.unknown?
```

Comment:
Changed this to omit unknown state from history as it is likely not useful for end users and we have got the error log to track if it happens.